tinygrad/tensor.py

```diff
@@ -341,10 +341,11 @@ def repeat(self, repeats):
 
   # TODO: make this nicer with syntactic sugar in slice
   def chunk(self, num, dim):
-    slice_params = [[(0, s) for s in self.shape] for _ in range(num)]
-    for i,k in enumerate(range(0, self.shape[dim], self.shape[dim]//num)):
-      slice_params[i][dim] = (k, min(self.shape[dim], k+self.shape[dim]//num))
-    return [self.slice(p) for p in slice_params]
+    dim = (dim + self.ndim) if dim < 0 else dim
+    x = self.permute(*(i for i in range(self.ndim) if i != dim), dim)
```

Comment:
Why is this permuting?

Reply:
In order not to deal with slice_params, I moved the dimension that will be chunked to last dim to use the "..." in the for loop. May be there is a more optimized approach, if you want I can try to find it

Reply:
Yea, do it the right way